src/main/java/net/dv8tion/jda/api/hooks/AnnotatedEventManager.java

```diff
@@ -57,9 +57,13 @@ public class AnnotatedEventManager implements IEventManager
     @Override
     public void register(@Nonnull Object listener)
     {
+        if (listener.getClass().isArray())
+            throw new IllegalArgumentException("Tried to register an array (" + listener.getClass().getCanonicalName() + ") as an annotated event listener." +
+                    "You might want to cast the array into Object[]");
```

Comment:
I don't really understand what this is supposed to mean. Casting here wouldn't solve anything, would it?

Reply:
This is specifically for `addEventListeners`, not `register(object)`. The important difference here is varargs.